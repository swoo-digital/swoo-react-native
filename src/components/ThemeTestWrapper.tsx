import React from 'react';
import {ThemeProvider} from 'styled-components/native';
import {theme} from '../theme';

interface ThemeTestWrapperProps {
  children: React.ReactNode;
}

const ThemeTestWrapper = (props: ThemeTestWrapperProps) => (
  <ThemeProvider theme={theme}>{props.children}</ThemeProvider>
);

export default ThemeTestWrapper;
