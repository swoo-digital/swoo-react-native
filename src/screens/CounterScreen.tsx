import React from 'react';
import Box from '../components/Box';
import Counter from '../components/Counter';
import styled from 'styled-components/native';
import Button from '../components/Button';
import {useAppDispatch, useAppSelector} from '../redux/hooks';

const ScreenContainer = styled.SafeAreaView`
  flex: 1;
  justify-content: center;
  align-items: center;
  background: ${props => props.theme.colors.bg.navy};
`;

function CounterScreen() {
  const count = useAppSelector(state => state.counter.counter);
  const dispatch = useAppDispatch();

  const onPressAdd = () => {
    dispatch({type: 'counter/add'});
  };

  const onPressSubtract = () => {
    dispatch({type: 'counter/subtract'});
  };

  return (
    <ScreenContainer>
      <Box>
        <Button title="+1" onPress={onPressAdd} testID="addButton" />
        <Counter amount={count} testID="counter" />
        <Button title="-1" onPress={onPressSubtract} testID="subtractButton" />
      </Box>
    </ScreenContainer>
  );
}

export default CounterScreen;
