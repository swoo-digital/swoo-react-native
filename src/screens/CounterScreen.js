import React from 'react';
import {
  decreamentCounter,
  increamentCounter,
} from '../store/features/counterSlice';
import {useDispatch, useSelector} from 'react-redux';
import {Box} from '../components/Box';
import {Touchable} from '../components/Touchable';
import {BtnText, Text} from '../components/Text';
import {SafeAreaView} from 'react-native';
import {colors} from '../utils/colors';

const CounterScreen = () => {
  const dispatch = useDispatch();
  const {value} = useSelector(store => store.counter);

  return (
    <SafeAreaView style={{backgroundColor: colors.bg}}>
      <Box>
        <Touchable onPress={() => dispatch(increamentCounter())}>
          <BtnText>+1</BtnText>
        </Touchable>
        <Text>{value}</Text>
        <Touchable onPress={() => dispatch(decreamentCounter())}>
          <BtnText>-1</BtnText>
        </Touchable>
      </Box>
    </SafeAreaView>
  );
};

export default CounterScreen;
