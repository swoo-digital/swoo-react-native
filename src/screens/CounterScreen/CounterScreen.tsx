import React, {useState} from 'react';
import {Box} from '../../components/Box';
import {PressableButton} from '../../components/PressableButton';
import {Counter} from '../../components/Counter';

export const CounterScreen = () => {
  const [count, setCount] = useState(0);

  const increment = () => setCount(prev => prev + 1);

  const decrement = () => setCount(prev => prev - 1);

  return (
    <Box>
      <PressableButton title="- 1" onPress={increment} />
      <Counter>{count}</Counter>
      <PressableButton title="+ 1" onPress={decrement} />
    </Box>
  );
};
