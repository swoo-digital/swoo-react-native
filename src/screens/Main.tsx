import React from 'react';
import {Provider} from 'react-redux';
import {ThemeProvider} from 'styled-components/native';
import {theme} from '../assets/theme/theme';
import CounterScreen from './CounterScreen';
import {store} from '../redux/store';

const Main = () => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <CounterScreen />
      </Provider>
    </ThemeProvider>
  );
};

export default Main;
