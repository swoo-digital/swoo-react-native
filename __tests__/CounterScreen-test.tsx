import React from 'react';
import renderer, {act} from 'react-test-renderer';
import CounterScreen from '../src/screens/CounterScreen';
import {Counter} from '../src/screens/CounterScreen';
import ThemeTestWrapper from '../src/components/ThemeTestWrapper';

describe('<CounterScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeTestWrapper>
          <CounterScreen />
        </ThemeTestWrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('CounterScreen', () => {
  it('should increment the counter', () => {
    const tree = renderer.create(
      <ThemeTestWrapper>
        <CounterScreen />
      </ThemeTestWrapper>,
    );
    const instance = tree.root;
    const button = instance.findByProps({label: '+1'});
    act(() => {
      button.props.onPress();
    });
    expect(instance.findByType(Counter).props.children).toBe(1);
  });

  it('should decrement the counter', () => {
    const tree = renderer.create(
      <ThemeTestWrapper>
        <CounterScreen />
      </ThemeTestWrapper>,
    );
    const instance = tree.root;
    const button = instance.findByProps({label: '-1'});
    act(() => {
      button.props.onPress();
    });
    expect(instance.findByType(Counter).props.children).toBe(-1);
  });
});
