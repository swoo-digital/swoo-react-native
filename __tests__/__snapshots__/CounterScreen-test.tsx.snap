// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterScreen /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#004761",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 24,
          "display": "flex",
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "width": 160,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
            },
          ]
        }
      >
        -1
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#8CD6BD",
            "fontSize": 64,
            "fontWeight": "bold",
            "paddingVertical": 80,
          },
        ]
      }
    >
      0
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 24,
          "display": "flex",
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "width": 160,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
            },
          ]
        }
      >
        +1
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
